tests/Language/Java/InlineSpec.hs

```diff
@@ -52,6 +56,11 @@ spec = do
           _ :: JJObject <- [java| $obj1 |]
           return ()
 
+        it "Supports parameterized type synonyms" $ do
+          obj :: List ObjectClass <- [java| new Object() {} |]
```

Comment:
It is a separate issue then. inline-java should provide at least some way to build where this type cast errors are not forgiven. JNI makes the program behavior undefined when the programmer  gets the types wrong.